cpp/src/parquet/arrow/writer.cc

```diff
@@ -356,7 +366,10 @@ class FileWriterImpl : public FileWriter {
 
   std::shared_ptr<::arrow::Schema> schema() const override { return schema_; }
 
-  Status WriteTable(const Table& table, int64_t chunk_size) override {
+  template <typename T>
+  Status WriteBuffered(const T& batch, int64_t max_row_group_length);
```

Comment:
```suggestion
  Status WriteBufferedRowGroup(const T& batch, int64_t max_row_group_length);
```

Reply:
I guess `WriteBuffered` is better than `WriteBufferedRowGroup`, because it might write more than 1 row-groups, or not write a rowgroup?

Reply:
Make sense